proposals/0035-linalg-matrix.md

```diff
@@ -1534,35 +1464,48 @@ class Matrix {
   template <MatrixComponentType NewCompTy, MatrixUse NewUse = Use>
   typename hlsl::enable_if<Scope != MatrixScope::Thread,
                            Matrix<NewCompTy, M, N, NewUse, Scope>>::type
-  cast();
+  Cast();
 
   // Element-wise operations
   template <typename T>
   typename hlsl::enable_if<hlsl::is_arithmetic<T>::value &&
                                Scope != MatrixScope::Thread,
                            Matrix>::type
-  operator+=(T);
+  operator+=(T V) {
+    for (uint I = 0; I < Length(); ++I)
+      Set(I, Get(I) + V);
+    return this;
+  }
+
   template <typename T>
   typename hlsl::enable_if<hlsl::is_arithmetic<T>::value &&
                                Scope != MatrixScope::Thread,
                            Matrix>::type
-  operator-=(T);
+  operator-=(T V) {
+    for (uint I = 0; I < Length(); ++I)
+      Set(I, Get(I) - V);
+    return this;
+  }
+
   template <typename T>
   typename hlsl::enable_if<hlsl::is_arithmetic<T>::value &&
                                Scope != MatrixScope::Thread,
                            Matrix>::type
-  operator*=(T);
+  operator*=(T V) {
+    for (uint I = 0; I < Length(); ++I)
+      Set(I, Get(I) * V);
+    return this;
+  }
+
   template <typename T>
   typename hlsl::enable_if<hlsl::is_arithmetic<T>::value &&
                                Scope != MatrixScope::Thread,
                            Matrix>::type
-  operator/=(T);
-
-  // Apply a unary operation to each element.
-  template <UnaryOperation Op, MatrixScope ScopeLocal = Scope>
-  typename hlsl::enable_if<Scope != MatrixScope::Thread && ScopeLocal == Scope,
-                           Matrix>::type
-  ApplyUnaryOperation();
+  operator/=(T V) {
+    for (uint I = 0; I < Length(); ++I)
+      Set(I, Get(I) / V);
+    return this;
+  }
```

Comment:
I expect you're aware, but for the sake of those who may not be, these operators are not supported by HLSL2021.

Reply:
Yes, and with the current API shape we really have two viable options. We could just not include these APIs (since they're trivial to implement), or we could propose relaxing the overload restrictions in HLSL 202x.